src/py42/services/cases.py

```diff
@@ -0,0 +1,194 @@
+        assignee=None,
+        page_number=1,
+        page_size=items_per_page,
+        sort_direction="asc",
+        sort_key="number",
+        **kwargs
+    ):
+        """Gets all cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status, "OPEN" or "CLOSED". Defaults to None.
+            created_at (str, optional): Filter results by case creation time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            updated_at (str, optional): Filter results by last updated time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            subject (str, optional): Filter results based on User UID of a subject of a case. Defaults to None.
+            assignee (str, optional): Filter results based on User UID of an assignee of a case. Defaults to None.
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 500.
+            sort_direction (str, optional): The direction on which to sort the response, based on the corresponding sort key. "asc" or "desc" Defaults to "asc".
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number". Available options are
+             name, number, createdAt, updatedAt, status, assigneeUsername, subjectUsername
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of cases.
+        """
+
+        return get_all_pages(
+            self.get_page,
+            u"cases",
+            name=name,
+            status=status,
+            created_at=created_at,
+            updated_at=updated_at,
+            assignee=assignee,
+            subject=subject,
+            page_number=page_number,
+            page_size=page_size,
+            sort_dir=sort_direction,
+            sort_key=sort_key,
+            **kwargs
+        )
+
+    def get_case(self, case_number):
+        """Retrieve case details by case number.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        return self._connection.get("{}/{}".format(self._uri_prefix, case_number))
+
+    def export_summary(self, case_number):
+        """Download case summary in a PDF file.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        uri_prefix = "{}/{}/{}".format(self._uri_prefix, case_number, "export")
+        return self._connection.get(uri_prefix)
+
+    def update(
+        self, case_number, name="", subject="", assignee="", description="", findings=""
```

Comment:
Will this overwrite existing case details? If that is the case (pun intended), we should first check for existing details and use those if the user does not specify here.. We do similar things in other methods that overwrite on update.

I specifically ask because you are defaulting to `""` here. If possible, we should default to `None`. Is that possible? Or does this api actually require all values to be `""` when not specified.

Reply:
Waiting for access to api to confirm the behaviour.

Reply:
Changed.